### PR TITLE
Refs #34043 -- Added context managers to SeleniumTestCase for changing window size.

### PR DESCRIPTION
**Pull request by sarahboyce**

### CODE REVIEW

File: tests/admin_views/tests.py [hunk 1]
```diff
@@ -6489,14 +6489,10 @@ def test_hidden_fields_small_window(self):
         )
         self.selenium.get(self.live_server_url + reverse("admin:admin_views_story_add"))
         field_title = self.selenium.find_element(By.CLASS_NAME, "field-title")
-        current_size = self.selenium.get_window_size()
-        try:
-            self.selenium.set_window_size(1024, 768)
+        with self.small_screen_size():
```
Comment (felixxm) [inline tests/admin_views/tests.py:6492]: Can you use it also in `tests/admin_widgets/tests.py`?

File: django/test/selenium.py [hunk 2]
```diff
@@ -118,6 +132,21 @@ def setUpClass(cls):
...
+
+    @contextmanager
+    def small_screen_size(self):
+        with ChangeWindowSize(1024, 768, self.selenium):
+            yield
+
+    @contextmanager
+    def mobile_size(self):
+        with ChangeWindowSize(360, 800, self.selenium):
+            yield
```
[the remark this answers is not in the document]
Comment (sarahboyce) [reply django/test/selenium.py:148]: We need to access the selenium setup on the test class that we're testing 🤔 Still thinking about it

[the remark this answers is not in the document]
Comment (felixxm) [reply django/test/selenium.py:148]: It's set on a test class, so it should not be an issue:

https://github.com/django/django/blob/9fd3a0ffc8fb8e532b0d5717f99fb15aff175a6c/django/test/selenium.py#L114-L119

[the remark this answers is not in the document]
Comment (sarahboyce) [reply django/test/selenium.py:148]:
```
    with ChangeWindowSize(1024, 768, cls.selenium):
                                     ^^^^^^^^^^^^
AttributeError: type object 'SeleniumTestCase' has no attribute 'selenium'
```
I think `setUpClass` is not called for the `@SeleniumTestCase.small_screen_size()` decorator and even if it was, I'm not sure the selenium instance would be the same.

I could apply the same fancy `__init_subclass__` approach also here and have that we want to run the test in a certain mode be independent as to whether we want to take a screenshot

[the remark this answers is not in the document]
Comment (sarahboyce) [reply django/test/selenium.py:169]: eb65242c0005d97a1d82a3d2bc68e6ca650a8aaf like this maybe?

**Comment by ngnpope** [comment]
I'm curious about the sizes we have (knowing that they are what we currently have in the CSS) as is anything "desktop" still not at least 1080p these days? Also, 1024×768 feels like and ancient thing… 🤔

As such, do we mean "small screen" or are we more interested in, say, "tablet" to go with "mobile" and "desktop"? That said, maybe we're better off going the Bootstrap way of `sm`, `md`, `lg`, and `xl` or whatever?

**Comment by felixxm** [comment]
> I'm curious about the sizes we have (knowing that they are what we currently have in the CSS) as is anything "desktop" still not at least 1080p these days? Also, 1024×768 feels like and ancient thing… 🤔
> 
> As such, do we mean "small screen" or are we more interested in, say, "tablet" to go with "mobile" and "desktop"? That said, maybe we're better off going the Bootstrap way of `sm`, `md`, `lg`, and `xl` or whatever?

That's definitely a topic for a long discussion and a lot of work (if we will decide to change it), so not something that should block or is needed for this PR.

File: django/test/selenium.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (felixxm) [reply django/test/selenium.py:169]: @sarahboyce Thanks :+1: You become very fluent with the `__init_subclass__` trick :muscle: However, it's not worth doing, IMO. I reverted to the previous state.

**Pull request merged**